src/test/java/oracle/kubernetes/operator/create/CreateDomainInputsValidationTest.java

```diff
@@ -246,6 +247,32 @@ public void createDomain_with_missingWeblogicDomainStoragePath_failsAndReturnsEr
       failsAndPrints(paramMissingError(PARAM_WEBLOGIC_DOMAIN_STORAGE_PATH)));
   }
 
+  @Test
+  public void createDomain_with_invalidWeblogicDomainStorageReclaimPolicy_failsAndReturnsError() throws Exception {
+    String val = "invalid-storage-reclaim-policy";
+    assertThat(
+      execCreateDomain(newInputs().weblogicDomainStorageReclaimPolicy(val)),
+      failsAndPrints(invalidEnumParamValueError(PARAM_WEBLOGIC_DOMAIN_STORAGE_RECLAIM_POLICY, val)));
+  }
+
+  @Test
+  public void createDomain_with_weblogicDomainStorageReclaimPolicyDeleteAndNonTmoWeblogicDomainStoragePath_failsAndReturnsError() throws Exception {
+    assertThat(
+      execCreateDomain(newInputs().weblogicDomainStorageReclaimPolicy(STORAGE_RECLAIM_POLICY_DELETE).weblogicDomainStoragePath("/scratch")),
+      failsAndPrints(invalidRelatedParamValueError(PARAM_WEBLOGIC_DOMAIN_STORAGE_RECLAIM_POLICY, STORAGE_RECLAIM_POLICY_DELETE, 
+                                              PARAM_WEBLOGIC_DOMAIN_STORAGE_PATH, "/scratch")));
+  }
+
+  @Test
+  public void createDomain_with_weblogicDomainStorageReclaimPolicyDelete_and_tmpWeblogicDomainStoragePath_succeeds() throws Exception {
+    GeneratedDomainYamlFiles
+      .generateDomainYamlFiles(
+        newInputs()
+          .weblogicDomainStorageReclaimPolicy(STORAGE_RECLAIM_POLICY_DELETE)
+          .weblogicDomainStoragePath("/tmp/"))
+      .remove();
+  }
+
```

Comment:
Need to add:
createDomain_with_weblogicDomainStorageReclaimPolicyRecycle_succeeds

(You implicitly get 'retain' tested since it's the default value, thus other tests cover it).

Reply:
There is a test for "Recycle" in run.sh. I guess I can add one here too.

Reply:
I'd prefer that - I've tried to have the unit tests cover everything, v.s. relying on integration tests.